Cython/Utility/ExtensionTypes.c

```diff
@@ -67,24 +68,35 @@ static int __Pyx_PyType_Ready(PyTypeObject *t) {
         // For details, see https://github.com/cython/cython/issues/3603
         PyObject *ret, *py_status;
         int gc_was_enabled;
-        PyObject *gc = PyImport_Import(PYUNICODE("gc"));
-        if (unlikely(!gc)) return -1;
-        py_status = PyObject_CallMethodObjArgs(gc, PYUNICODE("isenabled"), NULL);
+        static PyObject *gc = NULL;
+        static PyObject *gc_isenabled = NULL, *gc_enable = NULL, *gc_disable = NULL;
```

Comment:
I think there's a longer-term ambition to strip out static state so things like module-reloading and multiple interpreters can work (https://github.com/cython/cython/pull/3704, https://github.com/cython/cython/pull/3704).

I wonder if it's worth living with the cost of import and getattr here to avoid having to work around it in future?

Reply:
True, however this shouldn't be incompatible with module reloading and I suspect larger changes will be needed to support static types with multiple interpreters anyway.

Reply:
IIRC, we already have "static" types as heap types, if needed. Taking a step backwards on something as broad as `PyType_Ready()` doesn't feel great.

Looks like I stopped working on #3611 last year and left it to bit rot. I should pick it up again to get it merged. It has the right macro guards for optimisations like this one.